src/maintainer/knowledge_base.rst

```diff
@@ -1028,9 +1028,147 @@ In order to qualify as a noarch python package, all of the following criteria mu
   ``dataclasses``.
 
+.. hint::
+  
+  You can build platform-specific ``noarch`` packages to include runtime requirements depending on the target OS.
+  See mini-tutorial below.
+
 If an existing python package qualifies to be converted to a noarch package, you can request the required changes
 by opening a new issue and including ``@conda-forge-admin, please add noarch: python``.
 
+.. _os_specific_noarch:
+
+Noarch packages with OS-specific dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It is possible to build ``noarch`` packages with runtime requirements that depend on the target OS
+(Linux, Windows, MacOS), regardless the architecture (amd64, ARM, PowerPC, etc). This approach
+relies on four concepts:
+
+1.  `Virtual packages <https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html>`__. 
+    Prefixed with a double underscore, they are used by conda to represent system properties as
+    constraints for the solver at install-time. We will use ``__linux``, ``__win`` or ``__osx``,
+    which are only present when the running platform is Linux, Windows, or MacOS, respectively.
+    ``__unix`` is present in both Linux and MacOS. Note that this feature is **only fully available
+    on conda 4.10 or above**.
+2.  Jinja ``{% if ... %}`` conditionals, which can be used to mimic platform selectors.
+3.  ``conda-forge.ymls``'s :ref:`noarch_platforms` option.
+4.  `conda-build variants <https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html>`__.
+    We can use ``conda_build_config.yaml`` to create a matrix build that depends on the
+    ``noarch_platforms`` values.
+
+The idea is to generate different noarch packages for each OS needing different dependencies.
+Let's say you have a pure Python package, perfectly eligible for ``noarch: python``, but on Windows
+it requires ``windows-only-dependency``. You might have something like:
+
+.. code-block:: yaml
+  :caption: recipe/meta.yaml (original)
+
+  name: package
+  source:
+    # ...
+  build:
+    number: 0
+  requirements:
+    # ...
+    run:
+      - python
+      - numpy
+      - windows-only-dependency  # [win]
+
+Being non-noarch, this means that the build matrix will include at least 12 outputs: three platforms,
+times four Python versions. This gets worse with arm64, aarch64 and ppc64le in the mix. We can get it down
+to two outputs if replace it with this other approach!
+
+.. code-block:: yaml+jinja
+  :caption: recipe/meta.yaml (modified)
+
+  name: package
+  source:
+    # ...
+  build:
+    number: 0
+    # You can include target_os in the build string for easier identification of builds
+    string: "{{ target_os }}_pyh{{ PKG_HASH }}_{{ PKG_BUILDNUM }}"
+    noarch: python
+  requirements:
+    # ...
+    run:
+      - python
+      - numpy
+      - __{{ target_os }}
+      {% if target_os == 'win' %}
+      - windows-only-dependency
+      {% endif %}
```

Comment:
Ok my main concern about not using `conda_build_config.yaml`, @isuruf: 

* The package hashes will be the same (see https://github.com/conda-forge/click-feedstock/pull/44), so the build string NEEDS to be prefixed with the platform value. If the user forgets about this, it will result in a filename collision. This should be checked by the linter, at the very least. I am unaware if sharing a hash is problematic for other cases.

Reply:
I share this concern and would prefer not using `string`. Explained in more detail below ( https://github.com/conda-forge/conda-forge.github.io/pull/1839#discussion_r1008679066 )

Reply:
See https://github.com/conda/conda-build/pull/4606